apps/fft/src/butterfly.fil

```diff
@@ -0,0 +1,88 @@
+/// Butterfly module using existential parameters.
+comp Butterfly[W, E, ?M=W-E-1]<'G: L>(
+  go: interface['G],
+  in0[2]: ['G, 'G+L] W, // real and imaginary parts of input
+  in1[2]: ['G, 'G+L] W,
+  twiddle[2]: ['G, 'G+L] W // twiddle factor
+) -> (
+  out0[2]: ['G+L, 'G+L+1] W,
+  out1[2]: ['G+L, 'G+L+1] W
+) with {
+  some L where L > 0;
+} where
+    W == E + M + 1,
+    E > 1,
+    M > 1,
+    L > 0
+{
+  // switches between the type of butterfly unit
+  let T = 1;
```

Comment:
We should eventually expose some way for Filament designs to access a global `config` object to do this kind of switching without needing to change the source code. Nothing actionable; just a passing thought.